contracts/savings/peripheral/SaveWrapper.sol

```diff
@@ -18,125 +19,205 @@ interface IBoostedSavingsVault {
+        _saveAndStake(_save, _vault, _amount, true);
     }
 
     /**
      * @dev 1. Mints an mAsset and then deposits to SAVE
+     * @param _mAsset       mAsset address
      * @param _bAsset       bAsset address
-     * @param _amt          Amount of bAsset to mint with
+     * @param _save         Save address
+     * @param _vault        Boosted Savings Vault address
+     * @param _amount       Amount of bAsset to mint with
      * @param _minOut       Min amount of mAsset to get back
-     * @param _stake        Add the imUSD to the Savings Vault?
+     * @param _stake        Add the imAsset to the Boosted Savings Vault?
      */
-    function saveViaMint(address _bAsset, uint256 _amt, uint256 _minOut, bool _stake) external {
+    function saveViaMint(
+        address _mAsset,
+        address _save,
+        address _vault,
+        address _bAsset,
+        uint256 _amount,
+        uint256 _minOut,
+        bool _stake
+    ) external {
+        require(_mAsset != address(0), "Invalid mAsset");
+        require(_save != address(0), "Invalid save");
+        require(_vault != address(0), "Invalid vault");
+        require(_bAsset != address(0), "Invalid bAsset");
+
         // 1. Get the input bAsset
-        IERC20(_bAsset).transferFrom(msg.sender, address(this), _amt);
+        IERC20(_bAsset).transferFrom(msg.sender, address(this), _amount);
+
         // 2. Mint
-        IMasset mAsset_ = IMasset(mAsset);
-        uint256 massetsMinted = mAsset_.mint(_bAsset, _amt, _minOut, address(this));
-        // 3. Mint imUSD and optionally stake in vault
-        _saveAndStake(massetsMinted, _stake);
-    }
+        uint256 massetsMinted = IMasset(_mAsset).mint(_bAsset, _amount, _minOut, address(this));
 
+        // 3. Mint imAsset and optionally stake in vault
+        _saveAndStake(_save, _vault, massetsMinted, _stake);
+    }
 
     /**
-     * @dev 2. Buys a bAsset on Uniswap with ETH then mUSD on Curve
-     * @param _amountOutMin  Min uniswap output in bAsset units
-     * @param _path          Sell path on Uniswap (e.g. [WETH, DAI])
-     * @param _minOutMStable Min amount of mUSD to receive
-     * @param _stake         Add the imUSD to the Savings Vault?
+     * @dev 2. Buys a bAsset on Uniswap with ETH, then mints imAsset via mAsset,
+     *         optionally staking in the Boosted Savings Vault
+     * @param _mAsset         mAsset address
+     * @param _save           Save address
+     * @param _vault          Boosted vault address
+     * @param _uniswap        Uniswap router address
+     * @param _amountOutMin   Min uniswap output in bAsset units
+     * @param _path           Sell path on Uniswap (e.g. [WETH, DAI])
+     * @param _minOutMStable  Min amount of mAsset to receive
+     * @param _stake          Add the imAsset to the Savings Vault?
      */
     function saveViaUniswapETH(
+        address _mAsset,
+        address _save,
+        address _vault,
+        address _uniswap,
```

Comment:
Theoretically this could work with SushiSwap, I think.

Reply:
Yep